docs/instrumenting.asciidoc

```diff
@@ -1,17 +1,15 @@
 [[getting-started]]
```

Comment:
How do you feel about the changes here? There is a lot going on [here](https://www.elastic.co/guide/en/apm/agent/go/current/instrumenting-source.html), and I think it'd be useful to break things up a little bit. These changes create a new page that shows, briefly, the three steps required to get started. Pages for modules and custom instrumentation are nested underneath.

<img width="1021" alt="Screen Shot 2019-11-27 at 1 42 13 PM" src="https://user-images.githubusercontent.com/5618806/69761556-92ed4080-111c-11ea-9512-985b6331f01d.png">

Reply:
This is much nicer, thanks.

I think we should probably pull Context Propagation out of Custom Instrumentation, and have it at the same level. There's two aspects to instrumentation: instrumenting individual components (e.g. HTTP server, database driver), and then "context propagation" - connecting up the dots.

Reply:
Thank you. I pulled context propagation out of instrumentation and added a note about it in the setup section.